lib/api/v3/users/users_api.rb

```diff
@@ -47,18 +47,20 @@ def user_transition(allowed)
               fail ::API::Errors::InvalidUserStatusTransition
             end
           end
-        end
 
-        resources :users do
-          helpers ::API::V3::Users::CreateUser
+          def authorize_user_cru_allowed
+            allowed_admin = current_user.admin? && (current_user.active? || current_user.is_a?(SystemUser))
+            allowed_user = current_user.active? && current_user.allowed_to_globally?(:add_user)
 
-          post do
-            authorize_admin
-            create_user(request_body, current_user)
+            authorize_by_with_raise(allowed_admin || allowed_user)
           end
+        end
+
+        resources :users do
+          post &::API::V3::Utilities::Endpoints::Create.new(model: User).mount
 
           get do
-            authorize_admin
+            authorize_user_cru_allowed
 
             query = ParamsToQueryService.new(User, current_user).call(params)
```

Comment:
Did you look into replacing this end point by mounting a default endpoint as well?